auslib/db.py

```diff
@@ -1329,32 +1354,34 @@ def bleet(table, type_, changed_by, query):
             body.append("Row to be inserted:")
             body.append(UTF8PrettyPrinter().pformat(query.parameters))
 
-        msg = MIMEText("\n".join(body), "plain")
-        msg["Subject"] = "%s to %s detected" % (type_, table.t.name)
-        msg["from"] = from_addr
-
+        subj = "%s to %s detected" % (type_, table.t.name)
+        send_email(relayhost, port, username, password, to_addr, from_addr,
+                   table, subj, body)
         table.log.debug("Sending change notification mail for %s to %s", table.t.name, to_addr)
-        try:
-            conn = SMTP()
-            conn.connect(relayhost, port)
-            conn.ehlo()
-            conn.starttls()
-            conn.ehlo()
-        except:
-            table.log.exception("Failed to connect to SMTP server:")
-            return
-        try:
-            if username and password:
-                conn.login(username, password)
-            conn.sendmail(from_addr, to_addr, msg.as_string())
-        except:
-            table.log.exception("Failed to send change notification:")
-        finally:
-            conn.quit()
-
     return bleet
 
 
+def make_change_notifier_for_read_only(relayhost, port, username, password, to_addr, from_addr):
+    def bleet(table, type_, changed_by, query):
+        body = ["Changed by: %s" % changed_by]
+        where = [c for c in query._whereclause.get_children()]
+        row = table.select(where=where)[0]
+        if row['read_only'] != query.parameters['read_only']:
+            body.append("Row(s) to be updated as follows:")
+            data = {}
+            data['name'] = UnquotedStr(repr(row['name']))
+            data['product'] = UnquotedStr(repr(row['product']))
+            data['read_only'] = UnquotedStr("%s ---> %s" %
+                                            (repr(row['read_only']),
+                                             repr(query.parameters['read_only'])))
+            body.append(UTF8PrettyPrinter().pformat(data))
+
+        subj = "%s to %s detected" % (type_, table.t.name)
```

Comment:
This subject should be different than the existing change notification mail so that we can quickly distinguish between the two.